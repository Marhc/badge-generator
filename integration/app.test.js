const puppeteer = require('puppeteer');
const { port } = require('../jest-puppeteer.config').server;

describe('App', () => {
  let browser;
  let page;

  /**
   * Get output code generated by the app.
   *
   * @returns {Promise<string>} A promise that resolves to the generated code.
   */
  function getOutputCode() {
    return page.evaluate(() => document.querySelector('pre').innerText);
  }

  /* eslint-disable jest/no-hooks */

  beforeAll(async () => {
    // Prevent "Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout"
    // error in Travis CI and potentially other CI environments.
    jest.setTimeout(30000);

    browser = await puppeteer.launch({
      dumpio: true,
    });
  });

  afterAll(() => browser.close());

  beforeEach(async () => {
    page = await browser.newPage();
    await page.emulate({
      viewport: {
        width: 1920,
        height: 1080,
      },
      userAgent: 'Puppeteer',
    });
    await page.goto(`http://localhost:${port}/`);
  });

  afterEach(() => page.close());

  /* eslint-enable jest/no-hooks */

  it('setting repository and enabling some services', async () => {
    // Basic test
    await expect(page).toMatch('Github Badge Generator');

    // Set repository
    await page.focus('[title="Repository"] input');
    await page.keyboard.type('amercier/badge-generator', {
      delay: 20,
    });

    // Select services
    await page.click('label[title="Travis CI"]');
    await page.click('label[title="Codecov"]');
    await page.click('label[title="David DM"]');
    await page.click('label[title="David DM (dev)"]');
    expect(await getOutputCode()).toMatchSnapshot();

    // Change style
    await page.click('label[title="Flat square"]');
    expect(await getOutputCode()).toMatchSnapshot();
  });
});
