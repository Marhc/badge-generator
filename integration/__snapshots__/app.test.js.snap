// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App setting repository and enabling some services 1`] = `
"[![Build Status](https://img.shields.io/travis/amercier/badge-generator/master.svg)](https://travis-ci.org/amercier/badge-generator)
[![Test Coverage](https://img.shields.io/codecov/c/github/amercier/badge-generator/master.svg)](https://codecov.io/github/amercier/badge-generator?branch=master)
[![Dependency Status](https://img.shields.io/david/amercier/badge-generator.svg)](https://david-dm.org/amercier/badge-generator)
[![devDependency Status](https://img.shields.io/david/dev/amercier/badge-generator.svg)](https://david-dm.org/amercier/badge-generator#info=devDependencies)
"
`;

exports[`App setting repository and enabling some services 2`] = `
"[![Build Status](https://img.shields.io/travis/amercier/badge-generator/master.svg?style=flat-square)](https://travis-ci.org/amercier/badge-generator)
[![Test Coverage](https://img.shields.io/codecov/c/github/amercier/badge-generator/master.svg?style=flat-square)](https://codecov.io/github/amercier/badge-generator?branch=master)
[![Dependency Status](https://img.shields.io/david/amercier/badge-generator.svg?style=flat-square)](https://david-dm.org/amercier/badge-generator)
[![devDependency Status](https://img.shields.io/david/dev/amercier/badge-generator.svg?style=flat-square)](https://david-dm.org/amercier/badge-generator#info=devDependencies)
"
`;
